android/modules/ui/src/js/listview.js

```diff
@@ -80,13 +80,32 @@ exports.bootstrap = function(Titanium) {
 	
 	//convert name of UI elements into a constructor function.
 	//I.e: lookup("Titanium.UI.Label") returns Titanium.UI.createLabel function
-	function lookup(name) {
-		var lastDotIndex = name.lastIndexOf('.');
-		var proxy = eval(name.substring(0, lastDotIndex));
-		if (typeof(proxy) == undefined) return;
+	function lookup(namespace) {
 
-		var proxyName = name.slice(lastDotIndex + 1);
-		return proxy['create' + proxyName];
+		// handle Titanium widgets
+		if (/^(Ti|Titanium)/.test(namespace)) {
+			const namespaceIndex = namespace.lastIndexOf('.'),
+				  proxyName = namespace.slice(namespaceIndex + 1),
+				  parentNamespace = namespace.substring(0, namespaceIndex),
+				  parentProxy = eval(parentNamespace);
+	
+			if (parentProxy) {
+				return parentProxy['create' + proxyName];
+			}
+	
+		// handle Alloy widgets
+		} else {
+			let widget = Module.main.require('/alloy/widgets/' + namespace + '/controllers/widget');
+			if (!widget) {
+					Module.main.require(namespace);
```

Comment:
I think the assignment is missing here? Also, this does not seem to work with native modules.